src/main/java/webserver/SocketHandler.java

```diff
@@ -11,17 +11,19 @@
     private PrintWriter out;
     private SocketIO io = new SocketIO();
     private Socket client;
+    private String directory;
 
-    public SocketHandler(Socket client) throws  IOException {
+    public SocketHandler(Socket client, String directory) throws  IOException {
         this.in = io.createSocketReader(client);
         this.out = io.createSocketWriter(client);
         this.client = client;
+        this.directory = directory;
     }
 
     public void run() {
         Request request = new Request(in);
         Controller controller = new Controller();
-        Routes routes = new Routes(controller);
+        Routes routes = new Routes(controller, directory);
```

Comment:
Is `directory` or even `routes` used anywhere else?  It might be more flexible and more direct to just pass in the `Router`

```java
SocketHandler handler = new SocketHandler(new Router(new Routes(controller, directory))); //Pass in the true dependency directly
handler.run();
```